File: LibraryManagementSystem/src/main/java/org/libmansys/Input.java
package org.libmansys;
import java.io.FileReader;
import java.io.IOException;
import java.util.List;

import com.google.gson.Gson;

public class Input {
    private List<Novel> novelsList;
    private List<Novel> novelsListSelected;

    public Input(List<Novel> novelsList, List<Novel> novelsListSelected) {
        this.novelsList = novelsList;
        this.novelsListSelected = novelsListSelected;
    }

    // novels_list
    public void inputNovel(int index, String novelId, String novelTitle, String novelUrl, String lastUpdate, List<Novel> lst) {
        Novel novel = new Novel();
        try {
            novel.setIndex(index);
            novel.setId(novelId);
            novel.setTitle(novelTitle);
            novel.setUrl(novelUrl);
            novel.setLastUpdate(lastUpdate);
        } catch (IllegalArgumentException e) {
            System.err.println(e.getMessage());
            return;
        }
        lst.add(novel);
    }

    public void loadNovelsList(String fileName, List<Novel> lst) {
        Gson gson = new Gson();
        try (FileReader reader = new FileReader(fileName)) {
            Novel[] novels = gson.fromJson(reader, Novel[].class);
            for (Novel novel : novels) {
                inputNovel(Integer.parseInt(novel.getIndex()), novel.getId(), novel.getTitle(), novel.getUrl(), novel.getLastUpdate(), lst);
            }
        } catch (IOException e) {
            System.err.println("Error reading from file: " + e.getMessage());
        }
    }
}

File: Python_ver/input.py
from classes.novel import Novel
import json

class Input:
    def __init__(self, novels_list, novels_list_selected):
        self.novels_list = novels_list
        self.novels_list_selected = novels_list_selected

    # novels_list
    def input_novel(self, index, novel_id, novel_title, novel_url, last_update, lst):
        novel = Novel()
        novel.set_index(index)
        novel.set_id(novel_id)
        novel.set_title(novel_title)
        novel.set_url(novel_url)
        novel.set_last_update(last_update)
        lst.append(novel)

    def load_novels_list(self, file_name, lst):
        #reset list
        lst.clear()
        with open(file_name, "r") as f:
            data = json.load(f)
            for novel in data:
                self.input_novel(novel["_Novel__index"], novel["_Novel__id"], novel["_Novel__title"], novel["_Novel__url"], novel["_Novel__last_update"], lst)